onmt/translate/translator.py

```diff
@@ -408,8 +425,13 @@ def unbottle(m):
 
             # (c) Advance each beam.
             for j, b in enumerate(beam):
-                b.advance(out[:, j],
-                          beam_attn.data[:, j, :memory_lengths[j]])
+                if self.copy_attn:
```

Comment:
Would love to not have this `if`, maybe beam_copy can default to None

Reply:
It is a great advice. I also think adding `if` is not elegant. I have tried the following but failed:
if `self.copy_attn` is false, I set the  `beam_copy` to `None`.
However, the input of advance function is `beam_copy.data[:, j]`. `None.data[:, j]` is illegal. It seems that we have to use `if` to feed different arguments to advance function.